### PR TITLE
feat(CLOUDDST-22844): add cosign tasks to pipelines

### PR DESCRIPTION
**Pull request by emilyzheng**
This commit adds a task for collecting options required by cosign and adds cosign tasks to release pipelines.

### CODE REVIEW

File: tasks/collect-cosign-params/collect-cosign-params.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        set -x
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+
+        if [ $(jq '.sign | has("cosign-secret")' $DATA_FILE) == false ] ; then
+            echo "No secret name provided via 'sign.cosign-secret' key in Data."
```
Comment (mmalina) [inline tasks/collect-cosign-params/collect-cosign-params.yaml]: I see we already use `.sign.configMapName` for something else: https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/rh-sign-image/rh-sign-image.yaml#L65

So maybe use `.sign.cosignSecretName` to be consistent with that naming?

Comment (emilyzheng) [reply tasks/collect-cosign-params/collect-cosign-params.yaml]: Renamed it.

Comment (mmalina) [reply tasks/collect-cosign-params/collect-cosign-params.yaml]: Thanks.

**Review by johnbieren — COMMENTED** [review]
General question, do we want both rh-sign-image and rh-sign-image-cosign in the pipelines going forward? Or should the former be removed in favor of the latter

**Comment by scoheb** [comment]
> General question, do we want both rh-sign-image and rh-sign-image-cosign in the pipelines going forward? Or should the former be removed in favor of the latter

we probably want both or now. Pulling images with signature verification only works now with simple signing (rh-sign-image). I am unsure of the user workflow when signatures created with cosign are available.

**Comment by mmalina** [comment]
I have another general question. Should this be optional, at least initially? As implemented right now, the moment it makes it to production, it will fail the release pipeline for everybody, unless they add the new sign.cosignSecretName parameter to their RPA and the secret is actually there with the right contents.

File: pipelines/rh-advisories/README.md [hunk 2]
```diff
@@ -22,6 +22,9 @@ the rh-push-to-registry-redhat-io pipeline.
 | taskGitUrl | The url to the git repo where the release-service-catalog tasks to be used are stored | Yes | https://github.com/konflux-ci/release-service-catalog.git |
 | taskGitRevision | The revision in the taskGitUrl repo to be used | No | - |
 
+## Changes in 0.13.0
+- Add tasks `collect-cosign-params` and `rh-sign-image-cosign` to sign images by cosign
```
Comment (mmalina) [inline pipelines/rh-advisories/README.md]: Maybe add a note saying that `rh-sign-image-cosign` is only run if sign.cosignSecretName is set in the data file?

(Same for the other pipeline.)

Comment (emilyzheng) [reply pipelines/rh-advisories/README.md]: Added.

Comment (mmalina) [reply pipelines/rh-advisories/README.md]: Thanks.

File: tasks/collect-cosign-params/collect-cosign-params.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  workspaces:
+    - name: data
+      description: Workspace to read and save files
+  results:
+    - name: cosign-secret-name
+      type: string
+      description: Name of secret which contains AWS_REGION, AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY and SIGN_KEY
+  steps:
+    - name: collect-params
+      image: quay.io/konflux-ci/release-service-utils:38c3bfd7479c86b832cba5b61f9bbde40c469393
```
Comment (mmalina) [inline tasks/collect-cosign-params/collect-cosign-params.yaml]: We currently have a [PR](https://github.com/konflux-ci/release-service-catalog/pull/473/files) open to update all image references to `quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f` . Can you use the same in all your image references as well? That way we don't have to worry about which PR gets merged first and if the other PR needs updating to also cover this.

Comment (emilyzheng) [reply tasks/collect-cosign-params/collect-cosign-params.yaml]: Done.

Comment (mmalina) [reply tasks/collect-cosign-params/collect-cosign-params.yaml]: Thanks!

**Review by mmalina — APPROVED** [review]

**Comment by mmalina** [comment]
Thanks for the work on this. I approved the PR. Now we need the e2e to pass - we don't have any coverage for this yet, but at least it will confirm that without setting `sign.cosignSecretName` it will be just skipped and not break anything. Once e2e passes, we can merge this. Also, @johnbieren might want to take a look one last time.

File: pipelines/rh-advisories/README.md (hunk 2 above)
Comment (johnbieren) [reply pipelines/rh-advisories/README.md]: The when condition is awesome, nicely done

**Pull request merged**